src/haz3lmenhir/Lexer.mll

```diff
@@ -0,0 +1,134 @@
+    | "infinity" | "neg_infinity" | "nan" | "epsilon_float" | "pi" | "max_int" | "min_int" | "is_finite" | "is_infinite" | "int_of_float" | "float_of_int" | "string_of_int" | "string_of_float" | "string_of_bool" | "int_of_string" | "float_of_string" | "bool_of_string" | "abs" | "abs_float" | "ceil" | "floor" | "exp" | "log" | "log10" | "sqrt" | "sin" | "cos" | "tan" | "asin" | "acos" | "atan" | "mod" | "string_length" | "string_compare" | "string_trim" | "string_concat" | "string_sub" { BUILTIN(Lexing.lexeme lexbuf)}
+    | "-." { MINUS_FLOAT }
+    | whitespace {token lexbuf }
+    | newline { advance_line lexbuf; token lexbuf}
+    | ints as i { INT (int_of_string i) }
+    | float as f { FLOAT (parse_float_string f )}
+    | string as s { STRING (String.sub s 1 (String.length s - 2)) }
+    | sexp_string as s { SEXP_STRING (String.sub s 1 (String.length s - 2)) }
+    | "true" { TRUE }
+    | "false" { FALSE }
+    | "let" { LET }
+    | "in" { IN }
+    | "end" { END }
+    | "fun" { FUN }
+    | "case" { CASE }
+    | "if" { IF }
+    | "then" { THEN }
+    | "else" { ELSE }
+    | "[" { OPEN_SQUARE_BRACKET }
+    | "]" { CLOSE_SQUARE_BRACKET }
+    | "{" { OPEN_CURLY }
+    | "}" { CLOSE_CURLY }
+    | "(" { OPEN_PAREN }
+    | ")" { CLOSE_PAREN }
+    | "->" { DASH_ARROW }
+    | "=>" { EQUAL_ARROW }
+    | "=" { SINGLE_EQUAL }
+    (* Int ops*)
+    | "+" { PLUS }
+    | "-" { MINUS }
+    | "*" { TIMES }
+    | "/" { DIVIDE }
+    | "**" {POWER}
+    | "==" { DOUBLE_EQUAL }
+    | "!=" { NOT_EQUAL }
+    | "<" { LESS_THAN}
+    | "<=" { LESS_THAN_EQUAL }
+    | ">" { GREATER_THAN }
+    | ">=" { GREATER_THAN_EQUAL }
+    (* Float ops *)
+    | "+." { PLUS_FLOAT }
+    | "*." { TIMES_FLOAT }
+    | "/." { DIVIDE_FLOAT }
+    | "**." {POWER_FLOAT}
+    | "==." { DOUBLE_EQUAL_FLOAT }
+    | "!=." { NOT_EQUAL_FLOAT }
+    | "<." { LESS_THAN_FLOAT}
+    | "<=." { LESS_THAN_EQUAL_FLOAT }
+    | ">." { GREATER_THAN_FLOAT }
+    | ">=." { GREATER_THAN_EQUAL_FLOAT }
+    (* String Ops *)
+    | "++" { STRING_CONCAT }
+    | "$==" { STRING_EQUAL }
+    (* Bool ops *)
+    | "&&" { L_AND }
+    | "||" { L_OR }
+    | "!" { L_NOT }
+    | "|" { TURNSTILE }
+    | "," { COMMA }
+    | ":" { COLON }
+    (* Types *)
+    | "Int" { INT_TYPE }
+    | "Float" { FLOAT_TYPE }
+    | "Bool" { BOOL_TYPE }
+    | "String" { STRING_TYPE }
+    | "Unknown" { UNKNOWN }
+    | "Internal" { INTERNAL }
+    (* DHExp Annotations *)
+    | "()" { UNIT }
+    (* Filters *)
+    | "pause" {PAUSE}
+    | "debug" {DEBUG}
+    | "hide" {HIDE}
+    | "eval" {EVAL}
+    (* Other *)
+    | ";" {SEMI_COLON}
+    | "test" {TEST}
+    | "::" { CONS }
+    | "@" {AT_SYMBOL}
+    | "?" {QUESTION}
+    | "_" {WILD}
+    | "fix" {FIX}
+    | "typfun" {TYP_FUN}
+    | "type" {TYP}
+    | "$" {DOLLAR_SIGN}
+    | "~" {TILDE}
+    | "t" {T_TYP} (* TODO These make it very easy to get an unparseable program *)
+    | "p" {P_PAT}
+    | "tp" {TP_TPAT}
+    | "e" {E_EXP}
```

Comment:
@green726 Is this intended to be the final version for these tokens. It's very easy to have a program not parse if you have a variable called e, t, p, tp. Let me know if you both already discussed this format with @cyrus- before I joined

Reply:
@7h3kk1d I don't believe I discussed this fomat with cyrus yet. Yeah, it's definitely not the best and we can certainly change it. Do you have any ideas for what you would like it to be?